[Gradient Compression] Disable test_ddp_hook_parity_powerSGD on Gloo backend

wayi1 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#58802 [Gradient Compression] Disable test_ddp_hook_parity_powerSGD on Gloo backend**

This test can only be re-enabled once https://github.com/facebookincubator/gloo/pull/309 is picked up by Gloo submodule.

Differential Revision: [D28623214](https://our.internmc.facebook.com/intern/diff/D28623214/)

Commit:
[Gradient Compression] Disable test_ddp_hook_parity_powerSGD on Gloo backend

This test can only be re-enabled once https://github.com/facebookincubator/gloo/pull/309 is picked up by Gloo submodule.

Differential Revision: [D28623214](https://our.internmc.facebook.com/intern/diff/D28623214/)

[ghstack-poisoned]

rohan-varma reviewed (approved):
forward fix for https://github.com/pytorch/pytorch/pull/58784

Commit:
Update on "[Gradient Compression] Disable test_ddp_hook_parity_powerSGD on Gloo backend"


This test can only be re-enabled once https://github.com/facebookincubator/gloo/pull/309 is picked up by Gloo submodule.

Differential Revision: [D28623214](https://our.internmc.facebook.com/intern/diff/D28623214/)

[ghstack-poisoned]
